### PR TITLE
Initial split of development documentation off end user docs

### PR DESCRIPTION
**Pull request by eloquence**
## Status

Ready for review

## Description of Changes

In scope of this PR:
- Removes non-development docs and assets
- Adds a light bit of structure to the development docs
- Updates links accordingly
- Simplifies versioning/build strategy

Not in scope:
- Major organizational changes or additions

## To test the modified Docker container

1. Run `docker build -f deploy/Dockerfile .` from the repository root.
2. If the image builds successfully, run `docker run -p 127.0.0.1:5080:5080 <image_id>` to spin up a container.
3. Visit the running container using `localhost:5080` in your browser.
4. Confirm that the documentation is built correctly, and that you can access the generated PDF file.

### CODE REVIEW

**Comment by eloquence** [comment]
(You should be able to run `make docs` from this branch and preview the post-split organization in that manner.)

**Comment by gonzalo-bulnes** [comment]
It seems I can't add this PR to my project board... might it be a permissions issue @eloquence?

**Comment by eloquence** [comment]
Yes, I think that's why - adding a project board is a write operation to the issue you're adding, so it requires write access to the repo. I've now added the following permissions:
- SD maintainers can write
- All staff can read/comment

**Comment by eloquence** [comment]
We've discussed this as well at the SD server hangout today and so far feedback on this proposal has been uniformly positive :). So I think this is ready for a first pass! After this is merged, I'll need to port over any relevant new commits from `securedrop-docs` until the split is final.

**Review by gonzalo-bulnes — COMMENTED** [review]
@eloquence Things look good in general:

- `make docs-lint` is clean
- `make docs` works as expected, **but** prints :tangerine: the following error:
   ```
   [W 220831 14:38:40 web:2239] 404 GET /development/_static/releases.css (127.0.0.1) 0.37ms
   ```

I think it's worth fixing the missing file, and/or figuring out why it's requested if it's not necessary, but besides that, I'd say :shipit:!

**Comment by eloquence** [comment]
Re: `releases.css`, you found a bug that also exists in the main docs build :). I've opened a PR to resolve here https://github.com/freedomofpress/securedrop-docs/pull/381, and once that's reviewed and merged, I'll port it over here.

**Comment by eloquence** [comment]
(Just a note that I've rebased `main` and updated this branch in line with the latest `docs` changes.)

**Comment by eloquence** [comment]
https://github.com/freedomofpress/securedrop-dev-docs/pull/2 is a first stab at incorporating the developer-focused SecureDrop Workstation documentation from the main README, I'll clean that up a bit more and promote that to ready once this one gets merged, but early feedback welcome!

**Comment by eloquence** [comment]
I've pushed a commit to also update the build/deploy logic to only build the latest version of this documentation (since this is developer documentation, we don't need the `stable`/`latest` complexity). I've also ensured PDF generation works in the build container, and added some notes to the test plan to verify that.

**Comment by eloquence** [comment]
(Rebased to pull in the CSS fix from https://github.com/freedomofpress/securedrop-docs/pull/381.)

File: README.md
Comment (legoktm) [inline README.md:4]: regarded*

**Review by legoktm — APPROVED** [review]
LGTM overall! I found one very minor typo which you can fix as part of this or in a followup.

File: README.md
Comment (eloquence) [reply README.md]: Fixed!

**Pull request merged**
